hyperopt/spark.py

```diff
@@ -476,7 +477,9 @@ def run_task_on_executor(_):
                     result = domain.evaluate(params, ctrl=None, attach_attachments=False)
                     yield result
                 except BaseException as e:
-                    yield e
+                    error_string = traceback.format_exc()
+                    logger.error(error_string)
+                    yield e(error_string)
```

Comment:
I am not familiar with the usage of `e(error_string)`. I tried the following:
```try:
    print(test)
except BaseException as e:
    raise e("add some msg")
```
and got `TypeError: 'NameError' object is not callable`

Reply:
Yes. It requires to use the Exception class. updated.